docs/docs/user-guide.md

```diff
@@ -223,27 +264,30 @@ public class NumberList<E extends @Nullable Number> implements List<E> {...}
 Now it *is* legal to write `NumberList<@Nullable Integer`>, since `@Nullable
 Integer` is assignable to the bound `@Nullable Number`. It's *also* legal to
 write `NumberList<Integer>`, since plain `Integer` is assignable to `@Nullable
-Number`. Inside `@NullMarked`, plain `Integer` means a reference to an actual
-`Integer` value, never null. That just means that the `@Nullable Number` can
-technically be null but in this case it never will be.
+Number`. Inside `@NullMarked`, plain `Integer` means the same thing as `@NonNull
+Integer`: a reference to an actual `Integer` value, never `null`. That just
+means that the values represented by `E` can be `null` on some other
+parameterization of `NumberList`, but not in an instance of
+`NumberList<Integer>`.
 
-Of course this assumes that `List` itself is written in a way that allows null:
+Of course this assumes that `List` itself is written in a way that allows
+nullable type arguments:
 
 ```java
 @NullMarked
 public interface List<E extends @Nullable Object> {...}
 ```
 
 If it were `interface List<E>` rather than `interface List<E extends @Nullable
-Object>` then `NumberList<E extends @Nullable Number> implements List<E>` would not
-be legal. That's because `interface List<E>` is short for `interface List<E
+Object>` then `NumberList<E extends @Nullable Number> implements List<E>` would
+not be legal. That's because `interface List<E>` is short for `interface List<E
 extends Object>`. Inside `@NullMarked`, plain `Object` means "`Object` reference
-that can't be null". The `<E extends @Nullable Number>` from NumberList would
-not be compatible with `<E extends Object>`.
+that can't be `null`". The `<E extends @Nullable Number>` from `NumberList`
+would not be compatible with `<E extends Object>`.
 
 The implication of all this is that every time you define a type variable like
-`E` you need to decide whether it can represent a `@Nullable` type. If it can,
-then it must have a `@Nullable` bound. Most often this will be `<E extends
+`E` you need to decide whether it can be sustituted with a `@Nullable` type. If
```

Comment:
This will get out of scope for the current PR, but

I think it's important to emphasize that

* appearances of `E` in signatures become inputs to the "substitution" process that derives what the effective signatures of the *type members* you'll be using are
* for appearances of `E` inside your method bodies *that never happens*; it never gets substituted with anything else, it simply *is what it is*. If the bound isn't non-null, then it has parametric nullness, and all that really matters about that is that that type (a) is null-producing and (b) is not null-consuming.

Reply:
I thought "substitution" was the term the JLS uses.